### PR TITLE
Design and develop new page for display endpoints in statistics page

### PR DESCRIPTION
**Pull request by sajithab**
jira - https://wso2.org/jira/browse/APPM-633

### CODE REVIEW

**Comment by dinusha-dilrukshi** [comment]
This was applied maually due to conflicts. comimt https://github.com/wso2/carbon-appmgt/commit/e6281dc65b5fdcd36b0a71447f6671fa0255e7a8

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java [hunk 1]
```diff
@@ -1437,6 +1435,68 @@ public static NativeArray jsFunction_getSubscribedAPIsByUsers(Context cx, Script
...
+
+        try {
+            /**
+             * http and https endpoint resolving by reading AppManagerConfiguration
+             */
+            AppManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
+            List<Environment> environments = config.getApiGatewayEnvironments();
+            String envDetails = "";
+            String httpEndpoint = null, httpsEndpoint = null, endpoint = null;
+            for(int i=0; i<environments.size(); i++){
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1461]: Replace with for each.
for (Environment environment : environments) {}

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java [hunk 2]
```diff
@@ -1437,6 +1435,68 @@ public static NativeArray jsFunction_getSubscribedAPIsByUsers(Context cx, Script
...
+                Environment environment = environments.get(i);
+                envDetails = environment.getApiGatewayEndpoint();
+                try {
+                    httpEndpoint = envDetails.split(",")[0];
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1465]: Include null check before calling split()

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java [hunk 3]
```diff
@@ -1437,6 +1435,68 @@ public static NativeArray jsFunction_getSubscribedAPIsByUsers(Context cx, Script
...
+                Iterator it = apiList.iterator();
+                int i = 0;
+                while (it.hasNext()) {
+                    NativeObject row = new NativeObject();
+                    Object appObject = it.next();
+                    WebApp app = (WebApp) appObject;
+                    APIIdentifier apiIdentifier = app.getId();
+                    row.put("name", row, apiIdentifier.getApiName());
+                    /**
+                     * this WebApp is for read the registry values
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1483]: Change the comment with correct formatting. Use // for line comments and Capitalize the first character.

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java [hunk 4]
```diff
@@ -1437,6 +1435,68 @@ public static NativeArray jsFunction_getSubscribedAPIsByUsers(Context cx, Script
...
+                     */
+                    WebApp tempApp = apiProvider.getAPI(apiIdentifier);
+                    row.put("endpoint", row, (tempApp.getTransports().equals("http") ? httpEndpoint : httpsEndpoint) +
+                            (app.getContext().startsWith("/") ? app.getContext() : "/" + app.getContext())
+                            + "/" + apiIdentifier.getVersion() + "/");
+                    myn.put(i, myn, row);
+                    i++;
+                }
+            }
+        } catch (Exception e) {
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1493]: Should not catch generic exceptions.

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java [hunk 5]
```diff
@@ -1437,6 +1435,68 @@ public static NativeArray jsFunction_getSubscribedAPIsByUsers(Context cx, Script
...
+            handleException("Error occurred while getting the APIs", e);
```
Comment (jenananthan) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1494]: Change APIs ->APPs

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java (hunk 1 above)
Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1461]: These code segments are not longer exists in the class. Logic had changed.

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java (hunk 2 above)
Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1465]: These code segments are not longer exists in the class. Logic had changed.

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java (hunk 3 above)
Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1483]: Fixed - https://github.com/wso2/carbon-appmgt/pull/140

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java (hunk 4 above)
Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1493]: Fixed - https://github.com/wso2/carbon-appmgt/pull/140

File: components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java (hunk 5 above)
Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:1494]: Fixed - https://github.com/wso2/carbon-appmgt/pull/140

**Pull request closed without merging**